citesphere/src/main/java/edu/asu/diging/citesphere/web/user/ShowPeopleController.java

```diff
@@ -0,0 +1,125 @@
+package edu.asu.diging.citesphere.web.user;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import edu.asu.diging.citesphere.core.exceptions.GroupDoesNotExistException;
+import edu.asu.diging.citesphere.core.exceptions.ZoteroHttpStatusException;
+import edu.asu.diging.citesphere.core.service.ICitationManager;
+import edu.asu.diging.citesphere.model.bib.ICitation;
+import edu.asu.diging.citesphere.model.bib.IPerson;
+import edu.asu.diging.citesphere.user.IUser;
+import edu.asu.diging.citesphere.model.bib.impl.CitationResults;
+import java.util.stream.Stream;
+
+@Controller
+public class ShowPeopleController {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Autowired
+    private ICitationManager citationManager;
+
+    Map<IPerson, List<ICitation>> personCitationSortedMap = new TreeMap<IPerson, List<ICitation>>();
+
+    @RequestMapping(value = "/auth/group/{zoteroGroupId}/items/showPeople")
+    public String showPeople(Model model, Authentication authentication, @PathVariable("zoteroGroupId") String groupId,
+            @RequestParam(defaultValue = "1", required = false, value = "page") String page,
+            @RequestParam(defaultValue = "title", required = false, value = "sort") String sort) {
+        Integer pageInt = 1;
+        try {
+            pageInt = new Integer(page);
+        } catch (NumberFormatException ex) {
+            logger.warn("Trying to access invalid page number: " + page);
+        }
+        IUser user = (IUser) authentication.getPrincipal();
+        CitationResults results;
+        try {
+            results = citationManager.getGroupItems(user, groupId, null, pageInt, sort);
```

Comment:
this only get the citations on one page, doesn't it? the person index should be for the whole collection.

I would try to do as much as you can through aggregations in the db, otherwise we'll easily run into memory issues (some groups have > 20,000 entries).